testbed/src/lib.rs

```diff
@@ -105,47 +82,39 @@ pub fn run<T, U>(mut test: T, mut data: Data<U>, name: &str, mut width: u32, mut
 
-        match event {
-            Event::Input(i) => {
-                match i {
-                    Input::Press(Button::Keyboard(Key::Return)) => running = !running,
-                    Input::Press(Button::Mouse(MouseButton::Left)) => {
-                        match grabbing {
-                            None => grabbing = try_grab(&mut data.world, mouse_position, dummy),
-                            Some(_) => {}
-                        }
-                    }
-                    Input::Release(Button::Mouse(MouseButton::Left)) => {
-                        ungrab(&mut data.world, &mut grabbing)
-                    }
-                    Input::Move(Motion::MouseCursor(x, y)) => {
-                        mouse_position = window_to_world(width, height, &data.camera, x, y);
-                        update_grab(&mut data.world, mouse_position, grabbing);
-                    }
-                    Input::Resize(w, h) => {
-                        width = w;
-                        height = h;
-                    }
-                    _ => (),
+        if let Some(button) = e.press_args() {
+            if button == Keyboard(Key::Return) {
+                running = !running;
+            } else if button == Mouse(MouseButton::Left) {
+                match grabbing {
+                    None => grabbing = try_grab(&mut data.world, mouse_position, dummy),
+                    Some(_) => {}
                 }
-                test.process_input(i, &mut data);
-            }
-            Event::Update(_) => {
-                if running { test.step(&mut data); }
             }
-            Event::Render(args) => {
-                encoder.reset();
-                gfx2d.draw(&mut encoder,
-                           &output_color,
-                           &output_stencil,
-                           args.viewport(),
-                           |c, g| {
-                               graphics::clear([0.3, 0.3, 0.3, 1.0], g);
-                               debug_draw(&mut data.world, data.draw_flags, transform, c, g);
-                           });
-                device.submit(encoder.as_buffer());
+        }
+        if let Some(button) = e.release_args() {
+            if button == Mouse(MouseButton::Left) {
+                ungrab(&mut data.world, &mut grabbing);
             }
-            _ => {}
+        }
+        e.mouse_cursor(|x, y| {
+            mouse_position = window_to_world(width, height, &data.camera, x, y);
+            update_grab(&mut data.world, mouse_position, grabbing);
+        });
+        test.process_input(&e, &mut data);
+
+        e.resize(|w, h| {
+            width = w;
+            height = h;
+        });
+
+        window.draw_2d(&e, |c, g| {
+            graphics::clear([0.3, 0.3, 0.3, 1.0], g);
+            debug_draw(&mut data.world, data.draw_flags, transform, c, g);
+        });
+
+        if let Some(u) = e.update_args() {
+            if running { test.step(&mut data, u.dt as f32); }
```

Comment:
If a testbed example is able to run at 60FPS at all times then the delta time will be equal to the old `TIME_STEP`. However, if the processing required for a frame is more than 16.66ms then delta time will be larger. Most game/physics/etc logic you want to work off delta time in some form so that when the framerate lowers, the game itself doesn't slow down as well. In saying that, I am pretty new to Piston so don't know all the intricacies of how Piston's delta time works so I might be missing something.